tests/test_client_functional.py

```diff
@@ -673,6 +673,29 @@ async def handler(request):
             await resp.content.read()
 
 
+async def test_read_timeout_on_send(aiohttp_client: Any) -> None:
+    payload = b"data"
+    sock_read_timeout = 0.1
+
+    async def gen_payload():
+        await asyncio.sleep(0.5)
+        yield payload
+
+    async def handler(request):
+        read_data = len(await request.read())
+        return web.Response(text=f"{read_data}")
+
+    app = web.Application()
+    app.router.add_put("/", handler)
+    client = await aiohttp_client(app)
+
+    timeout = aiohttp.ClientTimeout(total=None, sock_read=sock_read_timeout)
+    resp = await client.put("/", data=gen_payload(), timeout=timeout)
+    txt = await resp.text()
+    assert txt == f"{len(payload)}"
+    resp.close()
```

Comment:
```suggestion
async def test_read_timeout_on_write(aiohttp_client: Any) -> None:
    async def gen_payload() -> AsyncIterator[str]:
        # Delay writing to ensure read timeout isn't triggered before writing completes.
        await asyncio.sleep(0.5)
        yield b"foo"

    async def handler(request: web.Request) -> web.Response:
        return web.Response(body=await request.read())

    app = web.Application()
    app.router.add_put("/", handler)

    timeout = aiohttp.ClientTimeout(total=None, sock_read=0.1)
    client = await aiohttp_client(app)
    async with client.put("/", data=gen_payload(), timeout=timeout) as resp:
        result = await resp.read()  # Should not trigger a read timeout.
    assert result == b"foo"
```

Minor suggestions to make this more concise and clear.